charts/keycloak/templates/secrets.yaml

```
@@ -14,7 +14,11 @@ metadata:
   {{- range $key, $value := $values.labels }}
   {{- printf "%s: %s" $key (tpl $value $ | quote) | nindent 4 }}
   {{- end }}
+{{- if $values.type }}
+type: {{ $values.type }}
+{{- else }}
 type: Opaque
+{{- end }}
```

I'd suggest you simply this:
```suggestion
type: {{ default $values.type "Opaque" }}
```

---

Good idea. And I think it would be 
```
type: {{ default "Opaque" $values.type}}
```

---

You're right, of course 😉